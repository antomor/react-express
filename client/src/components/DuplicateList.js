import React, { Component } from 'react';
import PropTypes from 'prop-types';
import DuplicateItem from './DuplicateItem';
import { withStyles } from 'material-ui/styles';
import GridList from 'material-ui/GridList';

const styles = theme => ({
  root: {
    display: 'flex',
    flexWrap: 'wrap',
    justifyContent: 'space-around',
    overflow: 'hidden',
    backgroundColor: theme.palette.background.paper,
  },
  gridList: {
    width: 500,
    height: 450,
  },
  subheader: {
    width: '100%',
  },
});
class DuplicateList extends Component {
  render() {
      var items = this.props.items.map(x => {
        return <DuplicateItem key={x.key} item={x}/>
      });
      const { classes } = this.props;
    return (
      <div className={classes.root}>
        <GridList cellHeight={160} className={classes.gridList} cols={3}>
          {items}
          </GridList>
      </div>
    );
  }
}

DuplicateList.propTypes = {
  items: PropTypes.arrayOf(PropTypes.shape({
    key: PropTypes.string.isRequired,
    value: PropTypes.number.isRequired,
  }))
}

export default withStyles(styles)(DuplicateList);