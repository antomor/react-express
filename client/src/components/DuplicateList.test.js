import React from 'react';
import DuplicateList from './DuplicateList';
import renderer from 'react-test-renderer';

describe('DuplicateList', () => {
  it('renders', () => {
    const items = [
      {
        key: 'aa',
        value: 1
      },
      {
        key: 'bb',
        value: 10
      }
    ]
    const component = renderer.create(
      <DuplicateList items={items}/>,
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

});