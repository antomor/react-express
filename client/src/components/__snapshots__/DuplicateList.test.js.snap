// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DuplicateList renders 1`] = `
<div
  className="DuplicateList-root-1"
>
  <ul
    className="MuiGridList-root-4 DuplicateList-gridList-2"
    style={
      Object {
        "margin": -2,
      }
    }
  >
    <li
      className="MuiGridListTile-root-5"
    >
      <div
        className="MuiGridListTile-tile-6"
      >
        <div
          className="MuiPaper-root-9 MuiPaper-shadow2-13 MuiPaper-rounded-10"
        >
          <div
            className="MuiPaper-root-9 MuiPaper-shadow2-13 MuiPaper-rounded-10"
          >
            aa
          </div>
          <div
            className="MuiPaper-root-9 MuiPaper-shadow2-13 MuiPaper-rounded-10"
          >
            1
          </div>
        </div>
      </div>
    </li>
    <li
      className="MuiGridListTile-root-5"
    >
      <div
        className="MuiGridListTile-tile-6"
      >
        <div
          className="MuiPaper-root-9 MuiPaper-shadow2-13 MuiPaper-rounded-10"
        >
          <div
            className="MuiPaper-root-9 MuiPaper-shadow2-13 MuiPaper-rounded-10"
          >
            bb
          </div>
          <div
            className="MuiPaper-root-9 MuiPaper-shadow2-13 MuiPaper-rounded-10"
          >
            10
          </div>
        </div>
      </div>
    </li>
  </ul>
</div>
`;
